libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificatePinnerFactory.java

```diff
@@ -17,17 +17,21 @@
       put(Environment.CHINA, ChinaCertificatePins.CERTIFICATE_PINS);
     }
   };
+  private static final String COM_EVENTS = "events.mapbox.com";
+  private static final String CHINA_EVENTS = "events.mapbox.cn";
+  private CertificateBlacklist certificateBlacklist;
 
   /**
    * Based on http://square.github.io/okhttp/3.x/okhttp/okhttp3/CertificatePinner.html
    *
    * @return The CertificatePinner instance
    */
-  CertificatePinner provideCertificatePinnerFor(Environment environment) {
+  CertificatePinner provideCertificatePinnerFor(Environment environment, CertificateBlacklist certificateBlacklist) {
+    this.certificateBlacklist = certificateBlacklist;
```

Comment:
This is not necessary. We can pass in the `CertificateBlacklist` through `addCertificatesPins` and then through `removeBlacklistedPins`.